osquery/tables/networking/freebsd/process_open_sockets.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+ *  Copyright (c) 2015, Wesley Shields.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <sys/param.h>
+#include <sys/queue.h>
+#include <libprocstat.h>
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/tables/system/freebsd/procstat.h"
+
+namespace osquery {
+namespace tables {
+
+// Heavily inspired by procstat(1) on FreeBSD.
+std::pair<std::string, int>& sockaddr_to_pair(struct sockaddr_storage* sstor) {
+  char buffer[INET6_ADDRSTRLEN] = {0};
+  struct sockaddr_in6* sin6;
```

Comment:
Nit, mind initializing these to `nullptr`, it helps a lot with static analysis later on.